docs/decisions/0001-connexion-replacement.md

```diff
@@ -0,0 +1,602 @@
+
+* Good, because there is a lot of variety in how you setup your schema, although most require defining the JSON/YAML yourself.
+* Bad, because the non-JSON/YAML approaches don't appear to be the main purpose of this API, and little documentation/examples exists regarding their usage.
+* Bad, because it seems the approach for running using [Marshmallow for the schema](https://github.com/flasgger/flasgger#using-marshmallow-schemas) isn't valid anymore as the parameters that Flask takes in don't match the example. From reading various other docs, Flask seems to have had a major version update that changed it a bit in recent years, so that is likely the cause.
+
+### Flask Smorest
+
+[Flask Smorest](https://github.com/marshmallow-code/flask-smorest) (formerly flask-rest-api).
+
+<details>
+<summary>Example implementation (Note this uses Flask's Blueprint approach which several other approaches here also support):</summary>
+
+```py
+from datetime import date
+
+from flask import Flask
+from flask.views import MethodView
+from flask_smorest import Api, Blueprint, abort
+from marshmallow import Schema, fields
+from marshmallow.validate import Regexp
+
+app = Flask(__name__)
+app.config["API_TITLE"] = "Example API"
+app.config["API_VERSION"] = "v1"
+app.config["OPENAPI_VERSION"] = "3.0.2"
+app.config["OPENAPI_URL_PREFIX"] = "/"
+app.config["OPENAPI_SWAGGER_UI_PATH"] = "/swagger-ui"
+app.config["OPENAPI_SWAGGER_UI_URL"] = "https://cdn.jsdelivr.net/npm/swagger-ui-dist/"
+api = Api(app)
+
+
+class User(Schema):
+    name: str = fields.String(required=True)
+    phone_number: str = fields.String(
+        required=True,
+        validate=Regexp("^([0-9]|\*){3}\-([0-9]|\*){3}\-[0-9]{4}$"),
+        metadata={"example": "123-456-7890"},
+    )
+    date_of_birth: date = fields.Date(required=False)
+
+
+class UserOut(User):
+    user_id = fields.Integer()
+
+
+users = {
+    1: UserOut.from_dict(
+        {"name": "Bob Smith", "phone_number": "123-456-7890", "date_of_birth": date(2000, 1, 1)}
+    )
+}
+
+user_blp = Blueprint("user", "user", url_prefix="/user")
+
+
+@user_blp.route("/<int:user_id>")
+class UserById(MethodView):
+    @user_blp.response(200, UserOut)
+    def get(self, user_id: int):
+        if user_id not in users:
+            abort(404)
+
+        return users[user_id]
+
+
+@user_blp.route("/")
+class UserView(MethodView):
+    @user_blp.arguments(User)
+    @user_blp.response(201, UserOut)
+    def post(self, data: dict):
+        user = User.from_dict(data)
+        next_id = max(users.keys()) + 1
+        user.user_id = next_id
+
+        users[next_id] = user
+        return user
+
+
+api.register_blueprint(user_blp)
+if __name__ == "__main__":
+    app.run(port=8080)
+```
+
+</details>
+
+* Good, because the API uses Marshmallow for its data schema, which is a [well maintained library](https://github.com/marshmallow-code/marshmallow) - it's in the same project.
```

Comment:
Interesting that it's in the same org as marshmallow. But looks like currently even the maintainer of smorest acknowledges that it has some tech debt (https://github.com/apiflask/apiflask/discussions/14#discussioncomment-571898).